core/src/impl/Kokkos_Profiling_Interface.hpp

```diff
@@ -54,6 +54,17 @@
 #include <iostream>
 #include <cstdlib>
 
+// NOTE: in this Kokkos::Profiling block, do not define anything that shouldn't
+// exist should Profiling be disabled
+
+namespace Kokkos {
+namespace Profiling {
+enum DeviceType { Serial, OpenMP, Cuda, HIP, OpenMPTarget, HPX, ROCm, Threads };
+constexpr const size_t device_type_bits = 8;
+constexpr const size_t instance_bits    = 24;
```

Comment:
Is that 8+24=32 to match the size of `int32_t`?
Did you consider introducing a type alias in `Profiling::`?